bitswap/network/httpnet/httpnet.go

```diff
@@ -267,6 +267,12 @@ func New(host host.Host, opts ...Option) network.BitSwapNetwork {
 
 	c := &http.Client{
 		Transport: t,
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			// we do not follow redirects. Providers should keep
+			// announcements up to
+			// date. https://github.com/boxo/issues/862.
+			return http.ErrUseLastResponse
```

Comment:
https://pkg.go.dev/net/http states 
> ErrUseLastResponse can be returned by Client.CheckRedirect hooks to control how redirects are processed. If returned, the next request is not sent and the most recent response is returned with its body unclosed.

@hsanjuan  does this mean we need to manually close body in client code that called `client.Do`?

Reply:
Whops cafa7e2 now closes the body after reading.